### DISCUSSION TITLE
Finding Tangents on polygons

### DISCUSSION THREAD

**Discussion by renauddl** [locationtech/jts#889] [Ideas]
Hi, 

Congratulation for this nice lib. 

I need to compute tangents to polygons: 
Given two convex and non-overlapping polygons, find the 4 tangents. 

There are some smart algo about that (turning tangents with O(N) complexity,
but I could not find them in JTS, maybe I did not look at the right place?
 
I could go with convex hull, which would give two tangents out of 4...

Thank you again
--
Renaud De Landtsheer

**Comment by dr-jts** [comment]
There's no function for computing tangents in JTS.  Perhaps you can find some Java code or an algorithm to do this and adapt it to JTS structures.
